src/mbgl/map/tile_worker.cpp

```diff
@@ -89,20 +92,32 @@ TileParseResult TileWorker::parsePendingLayers() {
     }
 
     result.state = pending.empty() ? TileData::State::parsed : TileData::State::partial;
+
+    if (result.state == TileData::State::parsed) {
+        placeLayers();
+    }
+
     return std::move(result);
 }
 
+void TileWorker::placeLayers() {
+    redoPlacement(&placementPending, placementConfig);
+    for (auto &p : placementPending) {
+        p.second->swapRenderData();
+        insertBucket(p.first, std::move(p.second));
+    }
+}
```

Comment:
Should this do `placementPending.clear()` before exiting? I don't think there's an execution path where `placementPending` can get reused afterwards but before `parseAllLayers` clears it, and we might as well reclaim the memory in the meantime.